File: Data Structures & Algorithms/trapping-rain-water/submission-1.rs
impl Solution {
    pub fn trap(height: Vec<i32>) -> i32 {
        let mut left = 0;
        let mut right = height.len() - 1;

        let mut leftMax = 0;
        let mut rightMax = 0;

        let mut total = 0;

        while left<right {
            if height[left]<height[right] {
                if height[left]>=leftMax {
                    leftMax = height[left];
                } else {
                    total += leftMax - height[left];
                }
                left += 1;
            } else {
                if height[right]>=rightMax {
                    rightMax = height[right];
                } else {
                    total += rightMax - height[right];
                }
                right -= 1;
            }
        }

        total
    }
}

File: Data Structures & Algorithms/permutation-string/submission-1.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        ws = len(s1)
        s1 = list(s1)
        s1 = sorted(s1)
        left = 0

        while left < len(s2):
            s3 = list(s2)
            s3 = s3[left:left+ws]
            s3 = sorted(s3)
            if s1 == s3:
                return True
            left += 1
        return False